lib/notifications/display.dart

```diff
@@ -202,6 +205,55 @@ class NotificationDisplayManager {
     );
   }
 
+  static void _onRemoveFcmMessage(RemoveFcmMessage data) async {
+    assert(debugLog('notif remove zulipMessageIds: ${data.zulipMessageIds}'));
+
+    final groupKey = _groupKey(data);
+
+    var haveRemaining = false;
+    final activeNotifications = await ZulipBinding.instance.androidNotificationHost
+      .getActiveNotifications(desiredExtras: [kExtraZulipMessageId]);
+    for (final statusBarNotification in activeNotifications) {
+      if (statusBarNotification == null) continue; // TODO(pigeon) eliminate this case
+      final notification = statusBarNotification.notification;
+
+      // Sadly we don't get toString on Pigeon data classes: flutter#59027
+      assert(debugLog('  existing notif'
+        ' id: ${statusBarNotification.id}, tag: ${statusBarNotification.tag},'
+        ' notification: (group: ${notification.group}, extras: ${notification.extras}))'));
+
+      // Don't act on notifications that are for other Zulip accounts/identities.
+      if (notification.group != groupKey) continue;
+
+      // Don't act on the summary notification for the group.
+      if (statusBarNotification.tag == groupKey) continue;
+
+      final lastMessageIdStr = notification.extras[kExtraZulipMessageId];
```

Comment:
As an independent improvement, could you write a dartdoc for `kExtraZulipMessageId`? I found myself wondering what it was, while reading this.